python/mxnet/contrib/onnx/_import/import_onnx.py

```diff
@@ -132,6 +132,33 @@ def from_onnx(self, graph):
             out = out[0]
         return out, argDict, auxDict
 
+    def get_graph_metadata(self, graph):
+        """
+        Get the model metadata from a given onnx graph.
+        """
+        _params = set()
+        for tensor_vals in graph.initializer:
+            _params.add(tensor_vals.name)
+
+        input_data = []
+        for graph_input in graph.input:
+            shape = []
+            if graph_input.name not in _params:
+                for val in graph_input.type.tensor_type.shape.dim:
+                    shape.append(val.dim_value)
```

Comment:
shape = [val.dim_value for val in graph_input.type.tensor_type.shape.dim]

Reply:
are all these attrs `type` `tensor_type` `shape` `dim` guaranteed to be not `None` if graph_input is not `None` ?

Reply:
yes, it is part of the onnx graph definition. But I will put a new check and throw a model malformed exception here.